### PR TITLE
add new recipe for doebase v2.0.1

### PR DESCRIPTION
**Pull request by kenza12**
Add new recipe for **doebase** v0.0.2 : Basic Optimal DoE (Design of Experiments) routines for metabolic pathway design.

Development repository: https://github.com/pablocarb/doebase

Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by kenza12** [comment]
@conda-forge-admin, please ping team

File: recipes/doebase/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  script: {{ PYTHON }} -m pip install -vv .
+  binary_relocation: true
+  missing_dso_whitelist:
+    - /lib64/libc.so.6
+    - $RPATH/libxml2-8ed442f7.so.2.6.26
+    - $RPATH/libz-a147dcb0.so.1.2.3
+    - /lib64/libm.so.6
+    - /lib64/libdl.so.2
+    - lib/libstdc++.so.6
+    - lib/libgcc_s.so.1
```
Comment (ocefpaf) [inline recipes/doebase/meta.yaml:23]: Why are these needed?

**Comment by breakthewall** [comment]
Hi,

doebase has a dependency with pySBOL which is not available as a conda package. We are currently working on it but in the meantime we have hard copied sbol folder into dependent packages we are publishing. Being a C wrapper, pySBOL has dependencies with system libraries and some errors occur during the building on conda-forge resources while the package has been well tested on Linux, macOS and Windows platforms on GitHub resources.

I hope it answers well at your question and I let Kenza add some additional informations if needed.

Cheers,
Joan

File: recipes/doebase/meta.yaml (hunk 1 above)
Comment (kenza12) [reply recipes/doebase/meta.yaml:23]: Hi @ocefpaf  ,

this is the kind of errors returned by conda that @breakthewall told you about:

```
2021-11-24T13:30:34.8922236Z   ERROR (pysbol,lib/python3.7/site-packages/pysbol/lib/Linux/.libs/libxml2-8ed442f7.so.2.6.26): /lib64/libm.so.6 not found in packages, sysroot(s) nor the missing_dso_whitelist.
2021-11-24T13:30:34.8922861Z .. is this binary repackaging?
```
We could avoid it by adding these libraries in the **missing_dso_whitelist**. And we proceeded in the same way for these 2 other packages : https://github.com/conda-forge/staged-recipes/pull/17106  https://github.com/conda-forge/staged-recipes/pull/16673 .

Thank you for your work.

**Comment by kenza12** [comment]
@conda-forge-admin, please ping team

**Review by beckermr — CHANGES_REQUESTED** [review]
We cannot accept prebuilt binaries. The missing DSO whitelist needed means the package has them.

File: recipes/doebase/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  version: {{ version }}
+
+source:
+  url: https://github.com/pablocarb/doebase/archive/refs/tags/{{ version }}.tar.gz
+  sha256: 4c2be23c261dbf47c38d7c1097c8e30be108369a5a2c3f0c6ef6062dc9e06172
+
+build:
+  number: 0
+  script: {{ PYTHON }} -m pip install -vv .
+  skip: true  # [win or py != 37]
```
Comment (beckermr) [inline recipes/doebase/meta.yaml:15]: Are you sure you only want this pac kage to work for python 3.7?

Comment (kenza12) [reply recipes/doebase/meta.yaml:15]: Hi @beckermr ,
yes this package is only working with python 3.7, thank you.
